File: design-pattern-java/app/src/test/java/design/pattern/java/patterns/creational/factory_method/FactoryMethodTest.java
package design.pattern.java.patterns.creational.factory_method;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class FactoryMethodTest {
  @Test
  public void itShouldCreateSmallCarInstanceWhenSmallCarFactoryIsUsed() {
    CarFactory factory = new SmallCarFactory();
    Car car = factory.produceCar();
    Assertions.assertTrue(car instanceof SmallCar);
  }  
  @Test
  public void itShouldCreateBigCarInstanceWhenBigCarFactoryIsUsed() {
    CarFactory factory = new BigCarFactory();
    Car car = factory.produceCar();
    Assertions.assertTrue(car instanceof BigCar);
  }
}

File: design-pattern-jvm/app/src/test/kotlin/design/pattern/kotlin/patterns/creational/factory_method/FactoryMethodTest.kt
package design.pattern.kotlin.patterns.creational.factory_method

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class FactoryMethodTestKt {
    @Test
    fun itShouldCreateSmallCarInstanceWhenSmallCarFactoryIsUsed() {
        val factory: CarFactory = SmallCarFactory()
        val car = factory.produceCar()
        Assertions.assertTrue(car is SmallCar)
    }

    @Test
    fun itShouldCreateBigCarInstanceWhenBigCarFactoryIsUsed() {
        val factory: CarFactory = BigCarFactory()
        val car = factory.produceCar()
        Assertions.assertTrue(car is BigCar)
    }
}
